TCAT/Model/Route.swift

```diff
@@ -122,86 +122,83 @@ class Route: NSObject, Codable {
         endName = Constants.General.destination
     }
 
-        func formatDirections(start: String?, end: String?) {
-            startName = start ?? Constants.General.currentLocation
-            endName = end ?? Constants.General.destination
-
-            let first = 0
-            for (index, direction) in rawDirections.enumerated() where direction.type == .walk {
-                // Change walking direction name to name of location walking from
-                if index == first {
-                    direction.name = startName
-                } else {
-                    direction.name = rawDirections[index - 1].stops.last?.name ?? rawDirections[index - 1].name
-                }
+    func formatDirections(start: String?, end: String?) {
+        startName = start ?? Constants.General.currentLocation
+        endName = end ?? Constants.General.destination
+
+        let first = 0
+        for (index, direction) in rawDirections.enumerated() where direction.type == .walk {
```

Comment:
would be more functional to use a `filter` and `forEach` but you can keep this if its more readable